apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/MigratorApp.java

```diff
@@ -19,8 +41,92 @@
   public static void main(String[] args) {
     LOGGER.info("Successfully started");
-    LOGGER.info("Hello, World!");
+
+    AppConfiguration appConfig = null;
+    try {
+      appConfig = AppConfiguration.readConfigFromEnvironmentVariables();
+      LOGGER.info("Application configured: '{}'", appConfig);
+    } catch (AppConfigurationException e) {
+      System.err.println(e.getMessage());
+      LOGGER.warn("Invalid app configuration.", e);
+      System.exit(EXIT_CODE_BAD_CONFIG);
+    }
+
+    MetricRegistry appMetrics = setupMetrics(appConfig);
+
+    // Create a data source for use by createOrUpdateSchema
+    HikariDataSource pooledDataSource =
+        createPooledDataSource(appConfig.getDatabaseOptions(), appMetrics);
+
+    // run migration
+    boolean migrationSuccess =
+        DatabaseSchemaManager.createOrUpdateSchema(pooledDataSource, appConfig);
+
     LOGGER.info("Shutting down");
-    System.exit(0);
+    if (!migrationSuccess) {
+      System.exit(EXIT_CODE_FAILED_MIGRATION);
+    }
+    System.exit(EXIT_CODE_SUCCESS);
+  }
+
+  /**
+   * Sets the metrics.
+   *
+   * @param appConfig the app config
+   * @return the metrics
+   */
+  private static MetricRegistry setupMetrics(AppConfiguration appConfig) {
+    MetricRegistry appMetrics = new MetricRegistry();
+    appMetrics.registerAll(new MemoryUsageGaugeSet());
+    appMetrics.registerAll(new GarbageCollectorMetricSet());
+    Slf4jReporter appMetricsReporter =
+        Slf4jReporter.forRegistry(appMetrics).outputTo(LOGGER).build();
+
+    MetricOptions metricOptions = appConfig.getMetricOptions();
+    if (metricOptions.getNewRelicMetricKey().isPresent()) {
+      SenderConfiguration configuration =
+          SenderConfiguration.builder(
+                  metricOptions.getNewRelicMetricHost().orElse(null),
+                  metricOptions.getNewRelicMetricPath().orElse(null))
+              .httpPoster(new OkHttpPoster())
+              .apiKey(metricOptions.getNewRelicMetricKey().orElse(null))
+              .build();
+
+      MetricBatchSender metricBatchSender = MetricBatchSender.create(configuration);
+
+      Attributes commonAttributes =
+          new Attributes()
+              .put("host", metricOptions.getHostname().orElse("unknown"))
+              .put("appName", metricOptions.getNewRelicAppName().orElse(null));
+
+      NewRelicReporter newRelicReporter =
+          NewRelicReporter.build(appMetrics, metricBatchSender)
+              .commonAttributes(commonAttributes)
+              .build();
+
+      newRelicReporter.start(metricOptions.getNewRelicMetricPeriod().orElse(15), TimeUnit.SECONDS);
+    }
+
+    appMetricsReporter.start(1, TimeUnit.HOURS);
+    return appMetrics;
+  }
+
+  /**
+   * @param dbOptions the {@link DatabaseOptions} to use for the application's DB
+   * @param metrics the {@link MetricRegistry} to use
+   * @return a {@link HikariDataSource} for the BFD database
+   */
+  private static HikariDataSource createPooledDataSource(
```

Comment:
feels like something that would go in bfd-shared-utils somewhere

Reply:
Added a TODO to handle this in 1558.